login/service.go

```diff
@@ -162,18 +161,29 @@ func (keycloak *KeycloakOAuthProvider) Perform(ctx *app.AuthorizeLoginContext, a
 	}
 
 	// First time access, redirect to oauth provider
+	redirect := ctx.Redirect
+	referrer := ctx.RequestData.Header.Get("Referer")
```

Comment:
My suggestion to handle this would be:
Till we work on #892 , hardcode the whitelist ( if possible? ) and then remove it as part of #892 
Saying this because not having *any* whitelist would create a hole when this code gets auto-deployed from master( after we merge ).